app/controllers/items_controller.rb

```diff
@@ -0,0 +1,28 @@
+class ItemsController < ApplicationController
+
+  def index
+      if params[:merchant_id].nil?
+        @items = Item.all
+      else
+        @merchant = Merchant.find(params[:merchant_id])
+        @items = @merchant.items
+      end
+    end
+
+  def show
+    @item = Item.find(params[:id])
+  end
+
+  def create
+    merchant = Merchant.find_by(params[:merchant_id])
```

Comment:
Here's where I was saying params[:merchant_id] is actually nil. Because you are using find_by rather than find, it's not throwing an error. It's essentially returning a random merchant.